crates/deploy/src/lib.rs

```diff
@@ -2,51 +2,174 @@ pub mod deploy_proto {
     include!("../proto/ambient.run.deploy.rs");
 }
 
-use std::{path::Path, str::FromStr};
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
 
 use ambient_project::Manifest;
+use md5::Digest;
+use tokio_stream::StreamExt;
 use tonic::{
     codegen::{CompressionEncoding, InterceptedService},
     metadata::MetadataValue,
+    service::Interceptor,
     transport::{Certificate, Channel, ClientTlsConfig, Uri},
     Request,
 };
 use walkdir::WalkDir;
 
-use deploy_proto::{deployer_client::DeployerClient, DeployAssetRequest, DeployAssetsResponse};
+use deploy_proto::{
+    asset_content::ContentDescription, deploy_asset_response::Message,
+    deployer_client::DeployerClient, AssetContent, DeployAssetRequest, VersionDeployed,
+};
+
+const CHUNK_SIZE: usize = 1024 * 1024 * 3; // 3MB
 
-async fn asset_from_file_path(
-    base_path: impl AsRef<Path>,
+async fn asset_requests_from_file_path(
+    ember_id: impl AsRef<str>,
+    asset_path: impl AsRef<str>,
     file_path: impl AsRef<Path>,
-) -> anyhow::Result<DeployAssetRequest> {
-    let path = file_path
-        .as_ref()
-        .strip_prefix(base_path)?
-        .to_string_lossy()
```

Comment:
I think failing here if a file path isn't UTF-8 would be preferred, since this can map different files to the same value, which can be a very nasty bug to figure out, especially for users who don't know this happens, as they think they did some configuring wrong.

Reply:
Agreed, better to fail early. I'll update it.